paleofetch.c

```diff
@@ -222,8 +223,27 @@ char *get_resolution() {
 }
 
 char *get_terminal() {
+    Display *display = XOpenDisplay(NULL);
+    unsigned char *prop;
+    unsigned long _, // not unused, but we don't need the results
+                  window = RootWindow(display, XDefaultScreen(display));
+    Atom a,
+         active = XInternAtom(display, "_NET_ACTIVE_WINDOW", True),
+         class = XInternAtom(display, "WM_CLASS", True);
+
+#define GetProp(property) \
+    XGetWindowProperty(display, window, property, 0, 64, 0, 0, &a, (int *)&_, &_, &_, &prop);
+
+    GetProp(active);
+    window = (prop[3] << 24) + (prop[2] << 16) + (prop[1] << 8) + prop[0];
```

Comment:
I would appreciate a comment explaining what this is doing here, its pretty opaque

Reply:
I messed up comment, this is referring to the `window =` line where you reverse the order of the bytes

Reply:
Well, it's essentially just to handle the disparity between how `XGetWindowProperty()` returns the property when the result is a window ID and what it expects as the second parameter (an "actual" window ID).

I'm sure there's a cleaner approach, but neither `htonl()` nor `ntohl()` does The Right Thing on my 64-bit machine. I concede that it's a hack, but I'm afraid I don't have a 32-bit system on hand to check whether it at least works there as well. But hey, 32-bit Arch is EOL anyway. :stuck_out_tongue: